Bottom: 205ff3a5f27670822fbd60af75e173d2249c1f84
Top:    dd8c303b04c065acc58e30c038edd2541b67b071
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-12 09:28:28 -0500

Add buildbot role and new playbook

---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
new file mode 100644
index 00000000..4d04b6bb
--- /dev/null
+++ b/ansible-playbook/buildbot.yml
@@ -0,0 +1,14 @@
+---
+
+- name: Install/Setup Docker images and containers
+  hosts: docker-hosts
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  roles:
+  - name: Build and start the Buildbot master container
+    role: buildbot-master
+    tags:
+    - buildbot-master
diff --git a/ansible-playbook/group_vars/all/buildbot.yml b/ansible-playbook/group_vars/all/buildbot.yml
new file mode 100644
index 00000000..86476751
--- /dev/null
+++ b/ansible-playbook/group_vars/all/buildbot.yml
@@ -0,0 +1,3 @@
+---
+buildbot_version: 1.1.2
+buildbot_worker_port: 9989
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index c306cadf..d476bf91 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -1,5 +1,6 @@
 ---
 # passwords
+buildbot_admin_password:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/admin"                      , length=20) }}'
 buildsystem_database_build_password: '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/readuser") }}'
 buildsystem_database_root_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/root"           , length=20) }}'
 buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/prod"                , length=20) }}'
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 23b1a3d9..b839e1e3 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -94,6 +94,9 @@
     tags:
     - buildsystem_container
 
+- name: Setup the buildbot containers
+  import_playbook: buildbot.yml
+
 - name: Setup the Build System (Test Station) with repo data
   hosts: buildsystem
   remote_user: ansibleremote
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
new file mode 100644
index 00000000..fa19756e
--- /dev/null
+++ b/roles/buildbot-master/defaults/main.yml
@@ -0,0 +1,52 @@
+---
+# defaults file for buildbot
+
+# The hostname passed as an envirnment variable into the container
+buildbot_ip_addr: '127.0.0.1'
+buildbot_hostname: buildbot.example.com
+buildbot_port_args: 
+  - '8080'
+  - '8443'
+  - '9989'
+
+# Network names
+buildbot_networks:
+  - name: '{{ docker_network_frontend }}'
+
+buildbot_data_path: '/buildbot'
+
+# the name of the image being duilt and used for the container
+buildbot_image_repo: https://github.com/novatechweb/docker-buildbot-master.git
+buildbot_image_name: '{{ docker_registry_username }}/buildbot'
+buildbot_image_dir: '{{ docker_projects_dir }}/buildbot-master'
+
+# the name of the container being started
+buildbot_container_name: '{{ docker_name_prefix }}buildbot'
+
+# the name of the data-volume used by the container
+buildbot_dv_name: '{{ docker_name_prefix }}buildbot_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+buildbot_docker_backup_dir: '{{ docker_backup_dir }}/buildbot'
+host_buildbot_docker_restore_dir: '{{ bacula_dest }}{{ buildbot_docker_backup_dir }}'
+
+buildbot_build_args:
+  BUILDBOT_UID: '1000'
+  BUILDBOT_VERSION: 'v1.1.2'
+  BUILDBOT_TIMEZONE: 'America/Chicago'
+
+buildbot_volumes:
+  - '{{ buildbot_dv_name }}:{{ buildbot_data_path }}:z'
+
+buildbot_env:
+  BUILDBOT_DATABASE: 'sqlite:///{{ buildbot_data_path }}/state.sqlite'
+  BUILDBOT_WEB_PORT: '8080'
+  BUILDBOT_WEB_URL: 'http://{{ buildbot_hostname }}/'
+  BUILDBOT_SMTP_HOST: 'relay.dnsexit.com'
+  BUILDBOT_SMTP_USER: 'Dean.cox'
+  BUILDBOT_SMTP_PASS: 'no5156'
+  BUILDBOT_WORKER_PORT: '9989'
+  BUILDBOT_WORKER_NTEL: 'worker-ntel'
+  BUILDBOT_WORKER_NTEL_PASS: '{{ buildbot_worker_ntel_password }}'
+  BUILDBOT_WORKER_PTXDIST: 'worker-ptxdist'
+  BUILDBOT_WORKER_PTXDIST_PASS: '{{ buildbot_worker_ptxdist_password }}'
diff --git a/roles/buildbot-master/env.yml b/roles/buildbot-master/env.yml
new file mode 100644
index 00000000..7ef0578f
--- /dev/null
+++ b/roles/buildbot-master/env.yml
@@ -0,0 +1,22 @@
+BUILDBOT_VERSION=1.1.1
+BUILDBOT_UID=1000
+
+BUILDBOT_VOL_CACHE=/home/coopera/Developer/buildbot-mounts/cache:/cache
+BUILDBOT_VOL_MASTER=/home/coopera/Developer/buildbot-mounts/buildbot:/buildbot
+BUILDBOT_VOL_NTEL=/home/coopera/Developer/buildbot-mounts/worker-ntel:/buildbot
+BUILDBOT_VOL_PTXDIST=/home/coopera/Developer/buildbot-mounts/worker-ptxdist:/buildbot
+BUILDBOT_VOL_SECRETS=./secrets:/run/secrets:ro
+
+BUILDBOT_DATABASE=sqlite:////cache/state.sqlite
+BUILDBOT_MASTER=buildbot
+BUILDBOT_TIMEZONE=America/Chicago
+BUILDBOT_HOST_PORT=172.16.103.71:80
+BUILDBOT_WEB_PORT=8010
+BUILDBOT_WEB_URL=http://buildbot.novatech-llc.com/
+
+BUILDBOT_WORKER_NTEL=worker-ntel
+BUILDBOT_WORKER_PASSWORD=pass
+BUILDBOT_WORKER_PORT=9989
+BUILDBOT_WORKER_PTXDIST=worker-ptxdist
+
+BUILDBOT_ENVIRONMENT_BLACKLIST=DOCKER_BUILDBOT*,BUILDBOT_ENV_*,BUILDBOT_1*,BUILDBOT_ENVIRONMENT_BLACKLIST,BUILDBOT_WORKER_PASSWORD
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
new file mode 100644
index 00000000..3415bab6
--- /dev/null
+++ b/roles/buildbot-master/tasks/main.yml
@@ -0,0 +1,40 @@
+---
+# tasks file for buildbot
+
+# This will generate passwords for these accounts.
+- assert:
+    that:
+    - buildbot_admin_password is defined
+
+- name: Create buildbot data volume
+  docker_volume:
+    name: '{{ buildbot_dv_name }}'
+
+- name: Checkout image repo
+  git:
+    repo: '{{ buildbot_image_repo }}'
+    version: master
+    dest: '{{ buildbot_image_dir }}'
+
+- name: Create buildbot master image
+  docker_image:
+    name: '{{ buildbot_image_name }}'
+    tag: '{{ docker_image_tag }}'
+    path: '{{ buildbot_image_dir }}'
+    buildargs: '{{ buildbot_build_args }}'
+    force: true
+
+- name: Start buildbot container
+  docker_container:
+    name: '{{ buildbot_container_name }}'
+    image: '{{ buildbot_image_name }}:{{ docker_image_tag }}'
+    networks: '{{ buildbot_networks }}'
+    volumes: '{{ buildbot_volumes }}'
+    env: '{{ buildbot_env }}'
+    ports: '{{ buildbot_port_args }}'
+
+- name: Add buildbot_master host
+  add_host:
+    name: buildbot_master
+    pubkey: "{{ output.stdout }}"
+    ip_addr: "{{ buildbot_ip_addr }}"
diff --git a/roles/buildbot-master/vars/main.yml b/roles/buildbot-master/vars/main.yml
new file mode 100644
index 00000000..eed3963e
--- /dev/null
+++ b/roles/buildbot-master/vars/main.yml
@@ -0,0 +1,2 @@
+---
+# vars file for buildbot
